--- configure.ac.old	2014-11-24 00:43:07.162984156 +0100
+++ configure.ac	2014-11-24 00:43:25.846984717 +0100
@@ -547,6 +547,7 @@ AC_CONFIG_FILES([Makefile
 		 src/plugins/select/bluegene/bl_bgq/Makefile
 		 src/plugins/select/bluegene/sfree/Makefile
 		 src/plugins/select/cons_res/Makefile
+		 src/plugins/select/cons_res_ext/Makefile
 		 src/plugins/select/cray/Makefile
 		 src/plugins/select/cray/libalps/Makefile
 		 src/plugins/select/cray/libemulate/Makefile
