*** slurm/src/slurmctld/job_mgr.c	2013-08-28 18:07:00.846965596 +0200
--- ../sim_test_dir/slurm/src/slurmctld/job_mgr.c	2012-11-29 15:26:35.000000000 +0100
***************
*** 474,479 ****
--- 474,480 ----
  	xstrcat(new_file, "/job_state.new");
  	unlock_slurmctld(job_read_lock);
  
+ #ifndef SLURM_SIMULATOR
  	if (stat(reg_file, &stat_buf) == 0) {
  		static time_t last_mtime = (time_t) 0;
  		int delta_t = difftime(stat_buf.st_mtime, last_mtime);
***************
*** 488,493 ****
--- 489,495 ----
  		}
  		last_mtime = time(NULL);
  	}
+ #endif
  
  	lock_state_files();
  	log_fd = creat(new_file, 0600);
***************
*** 3137,3148 ****
  
  	if (IS_JOB_RUNNING(job_ptr))
  		job_comp_flag = JOB_COMPLETING;
! 	else if (IS_JOB_PENDING(job_ptr))
  		job_ptr->start_time = now;
  
  	if ((job_return_code == NO_VAL) &&
  	    (IS_JOB_RUNNING(job_ptr) || IS_JOB_PENDING(job_ptr))) {
! 		info("Job %u cancelled from srun", job_ptr->job_id);
  	}
  
  	if (IS_JOB_SUSPENDED(job_ptr)) {
--- 3139,3152 ----
  
  	if (IS_JOB_RUNNING(job_ptr))
  		job_comp_flag = JOB_COMPLETING;
! 	else if (IS_JOB_PENDING(job_ptr)) {
! 		job_return_code = NO_VAL;
  		job_ptr->start_time = now;
+ 	}
  
  	if ((job_return_code == NO_VAL) &&
  	    (IS_JOB_RUNNING(job_ptr) || IS_JOB_PENDING(job_ptr))) {
! 		info("Job %u cancelled from interactive user", job_ptr->job_id);
  	}
  
  	if (IS_JOB_SUSPENDED(job_ptr)) {
